### PR TITLE
Activity pack editor/ view updates 

### PR DESCRIPTION
**Pull request by eadams17**
## WHAT
second round of updates to the activity pack editor for view/edit/create views

## WHY
these changes were requested by Curriculum

## HOW
a lot of refactoring to use Backpack and smaller sub-components as well as updating how requests are handled

### Screenshots
(If applicable. Also, please censor any sensitive data)
Screen Shot 2022-06-15 at 5 51 14 PM
Screen Shot 2022-06-15 at 5 51 43 PM
Screen Shot 2022-06-15 at 5 51 59 PM

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)
https://www.notion.so/quill/Activity-Editor-Updates-V2-f1c280e8b31c45f3a1961269ed4cc6e9

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  yes
Have you deployed to Staging? | yes
Self-Review: Have you done an initial self-review of the code below on Github? | yes
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | yes

### CODE REVIEW

File: services/QuillLMS/client/app/bundles/Teacher/components/unitTemplate/index.tsx [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+const SUCCESS_MESSAGE = 'Activity pack successfully saved!';
+
+export const UnitTemplate = ({ unitTemplate }) => {
+  const { id, flag, time, unit_template_category_id, name, activity_info, readability, diagnostic_names, activities } = unitTemplate;
+  const [activityPackFlag, setActivityPackFlag] = React.useState<DropdownObjectInterface>(flag ? {value: flag, label: flag} : null);
+  const [activityPackName, setActivityPackName] = React.useState<any>(name);
+  const [activityPackInfo, setActivityPackInfo] = React.useState<string>(activity_info);
+  const [activityPackType, setActivityPackType] = React.useState<DropdownObjectInterface>(null);
+  const [activityPackTime, setActivityPackTime] = React.useState<DropdownObjectInterface>(time ? {value: time.toString(), label: time.toString()} : null);
+  const [activityPackActivities, setActivityPackActivities] = React.useState<any>(activities ? activities : []);
```
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Teacher/components/unitTemplate/index.tsx]: `activities || []` is a little more succinct

File: services/QuillLMS/client/app/bundles/Teacher/components/unitTemplate/index.tsx [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+  function renderSubmissionModal() {
+    return(
+      <Modal className="rule-view-form-modal">
+        <section className="submission-modal-contents">
+          <p>{modalMessage}</p>
+          <button className="quill-button primary contained fun focus-on-light" onClick={handleCloseSubmissionModal}>Close</button>
+        </section>
+      </Modal>
+    )
+  }
```
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Teacher/components/unitTemplate/index.tsx]: I think this accomplishes the goal, but it seems like a snackbar would have accomplished the same purpose and been a little more in line with the design language we use elsewhere.

**Review by emilia-friedberg — APPROVED** [review]
This looks great. I appreciate the functional-ization of these components and the use of Backpack elements. One small thing - it was a little hard to tell from the spec what was actually supposed to have been accomplished in this PR vs what was done last time (I had to look at the history). It would be nice to either split up the tasks or outline in the PR notes which checkboxes apply to this one.

**Review by anathomical — APPROVED** [review]

File: services/QuillLMS/client/app/bundles/Teacher/components/unitTemplate/index.tsx (hunk 2 above)
Comment (eadams17) [reply services/QuillLMS/client/app/bundles/Teacher/components/unitTemplate/index.tsx]: good suggestion! I'll update to use a snackbar

**Pull request merged**